### DISCUSSION TITLE
how to get indexes and atoms with H  from smiles

### DISCUSSION THREAD

**Discussion by hal02** [rdkit/rdkit#6347] [Q&A]
hi all,

**I am trying to  add atom numbers in smiles as belows,**
#############
from rdkit import Chem
mol=Chem.MolFromSmiles('c1ccccc(C(N)=O)1')
for i, atom in enumerate(mol.GetAtoms()):
    atom.SetProp('molAtomMapNumber',str(i))
smi=Chem.MolToSmiles(mol)   
print(smi)
#############
**the output is:** 
[cH:0]1[cH:1][cH:2][cH:3][cH:4][c:5]1[C:6]([NH2:7])=[O:8]

**then I want to split the smiles into atoms, I did it like this:**

#############
from rdkit import Chem
mol=Chem.MolFromSmiles('c1ccccc(C(N)=O)1')
for i, atom in enumerate(mol.GetAtoms()):
    atom.SetProp('molAtomMapNumber',str(i))
    print(i,atom.GetSymbol())
#############
**the output is:**
0 C
1 C
2 C
3 C
4 C
5 C
6 C
7 N
8 O

**But what I do want is something like this:** 
0 cH
1 CH
...
7 NH2
8 O
 **I tried, but I cannot make it.**
**Can anyone help me figure out how to get each atom with H from the smiles as above.
thanks so much!**

**Comment by vfscalfani** [comment]
Hi @hal02, I'm not sure it is possible to do that directly without using some type of SMILES tokens, as the H's are generally not included in the graph. Then, if you add explicit H's like this, they are added at the end:



```
mol = Chem.MolFromSmiles('c1ccccc(C(N)=O)1')
molh = Chem.AddHs(mol)
for i, atom in enumerate(molh.GetAtoms()):
    atom.SetProp('molAtomMapNumber',str(i))
    print(i,atom.GetSymbol())
```
output:

```
0 C
1 C
2 C
3 C
4 C
5 C
6 C
7 N
8 O
9 H
10 H
11 H
12 H
13 H
14 H
15 H

```

One work around if you are just trying to print the output is to experiment with print statements and GetTotalNumHs(). Here is an example below. Note that this is not well tested or anything, so I'm not sure if it will quickly fail with other molecules:

```
mol = Chem.MolFromSmiles('c1ccccc(C(N)=O)1')
for i, atom in enumerate(mol.GetAtoms()):
    symbol = atom.GetSymbol()
    NumHs = atom.GetTotalNumHs()
    
    if NumHs == 0:
        print(symbol)

    elif NumHs == 1:
        print(symbol + "H")
    
    else:
        print(symbol + "H" + str(NumHs))

```

Output:

```
CH
CH
CH
CH
CH
C
C
NH2
O

```

Hope that helps, 

Vin
